core/conventional-commits/lib/recommend-version.js

```diff
@@ -51,14 +51,34 @@ function recommendVersion(pkg, type, { changelogPreset, rootPath, tagPrefix, pre
 
         // result might be undefined because some presets are not consistent with angular
         // we still need to bump _something_ because lerna saw a change here
-        const releaseType = data.releaseType || "patch";
+        let releaseType = data.releaseType || "patch";
 
         if (prereleaseId) {
           const shouldBump = shouldBumpPrerelease(releaseType, pkg.version);
           const prereleaseType = shouldBump ? `pre${releaseType}` : "prerelease";
           log.verbose(type, "increment %s by %s", pkg.version, prereleaseType);
           resolve(semver.inc(pkg.version, prereleaseType, prereleaseId));
         } else {
+          if (semver.major(pkg.version) === 0) {
+            // According to semver, major version zero (0.y.z) is for initial
+            // development. Anything MAY change at any time. The public API
+            // SHOULD NOT be considered stable. The version 1.0.0 defines
+            // the (initial stable) public API.
+            //
+            // To allow monorepos to use major version zero meaningfully,
+            // the transition from 0.x to 1.x must be explicitly requested
+            // by the user. Breaking changes MUST NOT automatically bump
+            // the major version from 0.x to 1.x.
+            //
+            // The usual convention is to use semver-patch bumps for bugfix
+            // releases and semver-minor for everything else, including
+            // breaking changes. This matches the behavior of `^` operator
+            // as implemented by `npm`.
+            //
+            if (releaseType === "major") {
+              releaseType = "minor";
```

Comment:
I guess if `--conventional-graduate` is `true`, we will bump the release to the new major, such as `0.1.0` -> `1.0.0`?

Reply:
Personally, I would find such rule as possibly confusing. Also, how would such rule work in large monorepos having packages both in pre-release versions (`2.0.0-beta.1`) and major zero versions (`0.2.5`)? 

I am currently envisioning the following process for graduating from 0.x to 1.0.0:

1. Add a commit changing the version in package.json to `1.0.0-1` (or a similar pre-release). This will be just a commit, no release/tag is associated with it. The commit can have a nice descriptive message for the changelog, e.g. `feat: graduate to 1.0`.

2. Run `lerna version` with `--conventional-graduate` to bump the temporary version `1.0.0-1` to final `1.0.0`.

Reply:
Please note `--conventional-graduate` allows a list of packages. Whether it's passed in as a CLI option or configured in `lerna.json`, we probably should honor it. See https://github.com/lerna/lerna/blob/master/commands/version/README.md#--conventional-graduate.

Reply:
I don't see `--conventional-graduate` affecting 0.x ranges, as they are not "prereleases" in the sense that option is intended to modify. 0.x ranges, as I understand them, mean "API unstable", not "this is a prerelease of a specified quality".